plugin-server/src/utils/db/postgres.ts

```diff
@@ -3,11 +3,143 @@
+        status.info('🤔', `Connecting to common Postgresql...`)
+        const commonClient = createPostgresPool(serverConfig.DATABASE_URL, serverConfig.POSTGRES_CONNECTION_POOL_SIZE)
+        status.info('👍', `Common Postgresql ready`)
+        // We fill the pools maps with the default client by default as a safe fallback for hobby,
+        // the rest of the constructor overrides entries if more database URLs are passed.
+        this.pools = new Map([
+            [PostgresUse.COMMON_WRITE, commonClient],
+            [PostgresUse.COMMON_READ, commonClient],
+            [PostgresUse.PLUGIN_STORAGE_RW, commonClient],
+        ])
+        this.statsd = statsd
+
+        if (serverConfig.DATABASE_READONLY_URL) {
+            status.info('🤔', `Connecting to read-only common Postgresql...`)
+            this.pools.set(
+                PostgresUse.COMMON_READ,
+                createPostgresPool(serverConfig.DATABASE_READONLY_URL, serverConfig.POSTGRES_CONNECTION_POOL_SIZE)
+            )
+            status.info('👍', `Read-only common Postgresql ready`)
+        }
+        if (serverConfig.PLUGIN_STORAGE_DATABASE_URL) {
+            status.info('🤔', `Connecting to plugin-storage Postgresql...`)
+            this.pools.set(
+                PostgresUse.PLUGIN_STORAGE_RW,
+                createPostgresPool(serverConfig.PLUGIN_STORAGE_DATABASE_URL, serverConfig.POSTGRES_CONNECTION_POOL_SIZE)
+            )
+            status.info('👍', `Plugin-storage Postgresql ready`)
+        }
+    }
+
+    public async query<R extends QueryResultRow = any, I extends any[] = any[]>(
+        target: PostgresUse | TransactionClient,
+        queryString: string | QueryConfig<I>,
+        values: I | undefined,
+        tag: string
+    ): Promise<QueryResult<R>> {
+        if (target instanceof TransactionClient) {
+            const wrappedTag = `${PostgresUse[target.target]}:Tx<${tag}>`
+            return postgresQuery(target.client, queryString, values, wrappedTag, this.statsd)
+        } else {
+            const wrappedTag = `${PostgresUse[target]}<${tag}>`
+            return postgresQuery(this.pools.get(target)!, queryString, values, wrappedTag, this.statsd)
+        }
+    }
+
+    public async bulkInsert<T extends Array<any>>(
+        usage: PostgresUse | TransactionClient,
+        // Should have {VALUES} as a placeholder
+        queryWithPlaceholder: string,
+        values: Array<T>,
+        tag: string
+    ): Promise<void> {
+        if (values.length === 0) {
+            return
+        }
+
+        const valuesWithPlaceholders = values
+            .map((array, index) => {
+                const len = array.length
+                const valuesWithIndexes = array.map((_, subIndex) => `$${index * len + subIndex + 1}`)
+                return `(${valuesWithIndexes.join(', ')})`
+            })
+            .join(', ')
+
+        await this.query(usage, queryWithPlaceholder.replace('{VALUES}', valuesWithPlaceholders), values.flat(), tag)
```

Comment:
I'm just surprised that our client library doesn't do this for us? (no action needed)

Reply:
agreed. just blindly copy-pasted from `db.ts` without any change